cpp/src/replace/nulls.cu

```diff
@@ -387,7 +387,7 @@ std::unique_ptr<cudf::column> replace_nulls_policy_impl(cudf::column_view const&
   auto valid_it  = cudf::detail::make_validity_iterator(*device_in);
   auto in_begin  = thrust::make_zip_iterator(thrust::make_tuple(index, valid_it));
 
-  rmm::device_vector<cudf::size_type> gather_map(input.size());
+  rmm::device_uvector<cudf::size_type> gather_map(input.size(), stream);
```

Comment:
I think you should also update line 405 below (cudf::detail::gather call) to pass it the same stream.